Source/SimCacheModule/Source/ViewModels/ViewModel.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#pragma once
+
+#include <Utils/Event/EventDispatcher.h>
+
+// -----------------------------------------------------------------------------
+
+class ViewModel
+{
+public:
+
+	ViewModel( Utils::EventDispatcher& ViewEventDispatcher );
+
+	virtual ~ViewModel();
+
+	virtual bool Initialize();
+	virtual void Uninitialize();
+
+protected:
+
+	Utils::EventDispatcher& GetViewEventDispatcher() const;
+
+private:
+
+	Utils::EventDispatcher& ViewEventDispatcher;
```

Comment:
Intentionally a ref this time! The event dispatcher is owned by the SimCacheModule and we have guarantees that the object will be alive for the duration of the view model.